### PR TITLE
Split persist new workspaces

### LINKED ISSUE #3654: Rename `PersistNewWorkspaces` state (or split it in two?)
Similar issue as #3653 

The state does more than just _persisting_ new workspaces, it also resets the the since params repository.

It would be helpful if we had a more general name (but still short) which would describe what the state does in the transitions diagram.

screen shot 2018-11-05 at 4 23 43 pm

Some of my ideas:
- `HandleNewWorkspaces` - too vague?
- `PersistNewWorkspacesAndResetSinceParams` - too long..

It might be better to split this in two states and first rest since params storage and then persist the new workspaces? 🤔 Maybe it's better than giving this state a confusing name. It would be similar to the `ScheduleCleanUp` state.

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
Splits the state to persist the new workspaces into two states, one that resets the since params and other that actually persist the new workspaces.

### Relationships

Closes #3654 

## Why do we want this?
The sync graph looks better, single responsibility, etc.

## How is it done?
GOTO What's this?

### Side effects
None?

## Review hints
Review the code and the sync graph!

## :squid: Permissions
🦑

### CODE REVIEW

**Comment by juanlaube** [comment]
[sync-graph.svg.zip](https://github.com/toggl/mobileapp/files/2554675/sync-graph.svg.zip)

**Comment by juanlaube** [comment]
screen shot 2018-11-06 at 4 18 42 pm

**Comment by juanlaube** [comment]
Tests.Sync in Bitrise: https://app.bitrise.io/build/0eff341376c8292c

File: Toggl.Foundation.Tests/Sync/States/Pull/ResetSinceParamsStateTests.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            Action tryingToConstructWithNulls = () => new ResetSinceParamsState(sinceParameterRepository);
+
+            tryingToConstructWithNulls.Should().Throw<ArgumentNullException>();
+        }
+
+        [Fact, LogIfTooSlow]
+        public async Task ResetsSinceParameterRepositoryBeforePersisting()
+        {
+            var state = new ResetSinceParamsState(sinceParameterRepository);
+            await state.Start(new IWorkspace[] { });
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/Sync/States/Pull/ResetSinceParamsStateTests.cs]: AAA

File: Toggl.Foundation.Tests/Sync/States/Pull/ResetSinceParamsStateTests.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using NSubstitute;
+using Toggl.Foundation.Sync.States.Pull;
+using Toggl.Foundation.Tests.Generators;
+using Toggl.Multivac.Models;
+using Toggl.PrimeRadiant;
+using Xunit;
+
+namespace Toggl.Foundation.Tests.Sync.States.Pull
+{
+    public class ResetSinceParamsStateTests
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/Sync/States/Pull/ResetSinceParamsStateTests.cs]: Seal please

File: Toggl.Foundation.Tests/Sync/States/Pull/ResetSinceParamsStateTests.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        [Fact, LogIfTooSlow]
+        public async Task ResetsSinceParameterRepositoryBeforePersisting()
+        {
+            var state = new ResetSinceParamsState(sinceParameterRepository);
+            await state.Start(new IWorkspace[] { });
+
+            sinceParameterRepository.Received().Reset();
+        }
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/Sync/States/Pull/ResetSinceParamsStateTests.cs]: Please add a simple test that it passes the very same list of workspaces without any change to the next state.

**Pull request merged**